app/controllers/games_controller.rb

```diff
@@ -13,14 +13,32 @@ def new
 	# keep pieces in correct places in database (coordinates)
 
 	def create
-		@game = current_user.games.create(game_create_params)
-		if @game.valid?
+		@game = current_user.games.new(game_create_params)
+		if params[:white_id].present?
+			@game.white_id = current_user.id 
+		else
+			@game.black_id = current_user.id
+		end
+
+		if @game.save
 			redirect_to game_path(@game)
 		else
-			redirect_to root_path, alert: "Could not create game."
+			flash[:error] = @game.errors.full_messages.join(", ")
+			render :new
```

Comment:
Just curious as to what the message actually says, I have never seen this implementation before